lib/ansible/plugins/inventory/foreman.py

```diff
@@ -175,7 +175,7 @@ def _get_all_params_by_id(self, hid):
         url = "%s/api/v2/hosts/%s" % (self.foreman_url, hid)
         ret = self._get_json(url, [404])
         if not ret or not isinstance(ret, MutableMapping) or not ret.get('all_parameters', False):
-            ret = []
+            ret = {}
```

Comment:
Shouldn't it be `return {}` though? Otherwise the line below will return `None`, and line 224 will fail since None is not iterable.

Reply:
Yes, you're quite right. Update incoming.